contracts/stabilizer/PSMRouter.sol

```diff
@@ -1,23 +1,60 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
 pragma solidity ^0.8.4;
 
 import "./IPSMRouter.sol";
 import "./PegStabilityModule.sol";
 import "../Constants.sol";
+import "../utils/RateLimited.sol";
+import "../pcv/IPCVDepositBalances.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 
+/// @notice the PSM router is an ungoverned, non custodial contract that allows user to seamlessly wrap and unwrap their WETH
+/// for trading against the PegStabilityModule.
 contract PSMRouter is IPSMRouter {
+    using SafeERC20 for IERC20;
 
+    /// @notice reference to the PegStabilityModule that this router interacts with
     IPegStabilityModule public immutable override psm;
+    /// @notice reference to the FEI contract used. Does not reference core to save on gas
+    /// Router can be redeployed if FEI address changes
+    IFei public override immutable fei;
+    /// @notice mutex lock to prevent fallback function from being hit any other time than weth withdraw
+    bool public override redeemActive;
 
-    constructor(IPegStabilityModule _psm) {
+    constructor(IPegStabilityModule _psm, IFei _fei) {
         psm = _psm;
+        fei = _fei;
         IERC20(address(Constants.WETH)).approve(address(_psm), type(uint256).max);
+        _fei.approve(address(_psm), type(uint256).max);
     }
 
     modifier ensure(uint256 deadline) {
         require(deadline >= block.timestamp, "PSMRouter: order expired");
         _;
     }
 
+    // ----------- Public View-Only API ----------
+
+    /// @notice view only pass through function to get amount of FEI out with given amount of ETH in
+    function getMintAmountOut(uint256 amountIn) public override view returns (uint256 amountFeiOut) {
+        amountFeiOut = psm.getMintAmountOut(amountIn);
+    }
+
+    /// @notice view only pass through function to get amount of ETH out with given amount of FEI in
+    function getRedeemAmountOut(uint256 amountFeiIn) public override view returns (uint256 amountTokenOut) {
+        amountTokenOut = psm.getRedeemAmountOut(amountFeiIn);
+    }
+
+    /// @notice the maximum mint amount out
+    function getMaxMintAmountOut() external view override returns(uint256) {
+        return psm.getMaxMintAmountOut();
+    }
```

Comment:
This is for what?

Reply:
This is for the UI to make Seb's life easier.

Reply:
If the router didn't have this passthrough, then the UI would have to hit multiple contracts to coordinate a swap. Now, it can just hit the PSMRouter which has a lot of the same view functions that just pass their calls to the PSM and then return.